frappe/integrations/doctype/nextcloud_settings/nextcloud_settings.py

```diff
@@ -0,0 +1,207 @@
+			if response_public_file == 'Failed': 
+				did_not_upload.append(public_file_backup)
+				error_log.append('Failed while uploading Public files')
+		if private_file_backup:
+			response_private_file = upload_backup(session, url, private_file_backup)
+			if response_private_file == 'Failed': 
+				did_not_upload.append(private_file_backup)
+				error_log.append('Failed while uploading Private files')
+
+	def prepare_backup(self, db_backup=None, site_config=None, public_file_backup=None, private_file_backup=None):
+		if frappe.flags.create_new_backup:
+			backup = new_backup(ignore_files= True if cint(self.backup_files) else False)
+			db_backup = os.path.join(self.backup_path, os.path.basename(backup.backup_path_db))
+			site_config = os.path.join(self.backup_path, os.path.basename(backup.backup_path_conf))
+			if cint(self.backup_files):
+				public_file_backup = os.path.join(self.backup_path, os.path.basename(backup.backup_path_files))
+				private_file_backup = os.path.join(self.backup_path, os.path.basename(backup.backup_path_private_files))
+
+		else:
+			if cint(self.backup_files):
+				db_backup, site_config, public_file_backup, private_file_backup = get_latest_backup_file(with_files=True)
+
+				if not public_file_backup or not private_file_backup:
+					generate_files_backup()
+					db_backup, site_config, public_file_backup, private_file_backup = get_latest_backup_file(with_files=True)
+			else:
+				db_backup, site_config,  = get_latest_backup_file()
+
+		return db_backup, site_config, public_file_backup, private_file_backup
+
+	def make_session(self):
+		session = requests.session()
+		session.verify = True
+		session.stream = True
+		session.auth = (self.email, self.get_password(fieldname='password',raise_exception=False))
+		session.headers.update({
+			"OCS-APIRequest": "true",
+		})
+		return session
+
+	def make_baseurl(self):
+		vurl = urlparse(self.nextcloud_url)
+		if not vurl.scheme:
+			return None
+		if not vurl.netloc:
+			return None
+		if not vurl.port:
+			port = 443 if vurl.scheme == 'https' else 80
+
+		base_url = '{0}://{1}:{2}'.format(vurl.scheme, vurl.netloc, vurl.port if vurl.port else port)
+		if self.webdav_url.startswith('/'):
+			base_url = '{0}{1}'.format(base_url, self.webdav_url)
+		else:
+			base_url = '{0}/{1}'.format(base_url, self.webdav_url)
+		if not base_url.endswith('/'):
+			base_url = '{0}/'.format(base_url)
+		return base_url
+
+	def check_for_upload_folder(self, session, url):
+		response = session.request("PROPFIND", url, headers={"Depth": "0"}, allow_redirects=False)
+		if response.status_code == 404:
+			if self.upload_path:
+				frappe.throw(_('Given "Path to upload folder" does not exist'))
+			else:
+				response = session.request("MKCOL", url, allow_redirects=False)
+				if response.status_code != 201:
+					frappe.throw(_('There was an error. Please try again'))
+
+def upload_backup(session, baseurl, filebackup):
```

Comment:
Any reason to leave it out of the class?

Reply:
It wasn't using any variable from the self anyway. So thought to kept it out